src/nvim/lua/treesitter.c

```diff
@@ -377,6 +385,86 @@ static int parser_edit(lua_State *L)
   return 0;
 }
 
+static int parser_set_ranges(lua_State *L)
+{
+  if (lua_gettop(L) < 2) {
+    return luaL_error(
+        L,
+        "not enough args to parser:set_included_ranges()");
+  }
+
+  TSLua_parser *p = parser_check(L);
+  if (!p || !p->tree) {
+    return 0;
+  }
+
+  if (!lua_istable(L, 2)) {
+    return luaL_error(
+        L,
+        "argument for parser:set_included_ranges() should be a table.");
+  }
+
+  size_t tbl_len = lua_objlen(L, 2);
+  TSRange *ranges = xmalloc(sizeof(TSRange) * tbl_len);
+
+
+  // [ parser, ranges ]
+  for (size_t index = 0; index < tbl_len; index++) {
+    lua_rawgeti(L, 2, index + 1);  // [ parser, ranges, range ]
+
+    if (!lua_istable(L, -1)) {
+      xfree(ranges);
+      return luaL_error(
+          L,
+          "argument for parser:set_included_ranges() should be a table of tables.");
+    }
+
+    if (lua_objlen(L, -1) < 2) {
```

Comment:
Probably `!=` rather than `<`

Reply:
Oh yeah of course... Ho can I miss that ...